src/binary-reader-linker.cc

```diff
@@ -243,11 +261,16 @@ Result BinaryReaderLinker::OnExport(Index index,
                                     ExternalKind kind,
                                     Index item_index,
                                     StringSlice name) {
+  if (kind == ExternalKind::Memory) {
+    WABT_FATAL("Linker does not support exported memories");
+  }
+
```

Comment:
Certainly not writing you off.  Your efforts are much appreciated.   On that note would you like to prepare a PR that contains just this FATAL error, along with on for memory imports?  It would be could to have such a thing weather of not this change eventually lands.